core/src/main/java/com/linecorp/armeria/server/RoutingResultBuilder.java

```diff
@@ -46,65 +46,65 @@
     @Nullable
     private MediaType negotiatedResponseMediaType;
 
-    RouteResultBuilder pathMappingResult(PathMappingResult pathMappingResult) {
+    RoutingResultBuilder pathMappingResult(PathMappingResult pathMappingResult) {
         this.pathMappingResult = requireNonNull(pathMappingResult, "pathMappingResult");
         return this;
     }
 
     /**
      * Sets the mapped path, encoded as defined in <a href="https://tools.ietf.org/html/rfc3986">RFC3986</a>.
      */
-    public RouteResultBuilder path(String path) {
+    public RoutingResultBuilder path(String path) {
         this.path = requireNonNull(path, "path");
         return this;
     }
 
     /**
      * Sets the specified query.
      */
-    public RouteResultBuilder query(@Nullable String query) {
+    public RoutingResultBuilder query(@Nullable String query) {
         this.query = query;
         return this;
     }
 
     /**
      * Sets the path parameters.
      */
-    public RouteResultBuilder pathParams(Map<String, String> pathParams) {
+    public RoutingResultBuilder pathParams(Map<String, String> pathParams) {
         this.pathParams = requireNonNull(pathParams, "pathParams");
         return this;
     }
 
     /**
      * Sets the score.
      */
-    public RouteResultBuilder score(int score) {
+    public RoutingResultBuilder score(int score) {
         this.score = score;
         return this;
     }
 
     /**
      * Sets the negotiated producible {@link MediaType}.
      */
-    public RouteResultBuilder negotiatedResponseMediaType(MediaType negotiatedResponseMediaType) {
+    public RoutingResultBuilder negotiatedResponseMediaType(MediaType negotiatedResponseMediaType) {
         this.negotiatedResponseMediaType = requireNonNull(negotiatedResponseMediaType,
                                                           "negotiatedResponseMediaType");
         return this;
     }
 
     /**
-     * Returns a newly-created {@link RouteResult}.
+     * Returns a newly-created {@link RoutingResult}.
      */
-    public RouteResult build() {
+    public RoutingResult build() {
         if (pathMappingResult != null) {
-            return new RouteResult(pathMappingResult, score, negotiatedResponseMediaType);
+            return new RoutingResult(pathMappingResult, score, negotiatedResponseMediaType);
```

Comment:
How about spreading the fields of `PathMappingResult`? i.e.

```java
new RoutingResult(path, query, pathParams, score, negotiatedResponseMediaType);
```

`RoutingResult` and `RoutingResultBuilder` will not need to keep `PathMappingResult` but just the three fields from it for reduced indirection.